print-esp.c

```diff
@@ -353,18 +367,13 @@ espprint_decode_encalgo(netdissect_options *ndo,
 	}
 	*colon = '\0';
 
-	if (strlen(decode) > strlen("-hmac96") &&
-	    !strcmp(decode + strlen(decode) - strlen("-hmac96"),
-		    "-hmac96")) {
-		p = strstr(decode, "-hmac96");
-		*p = '\0';
+	if (strendswith(decode, "-hmac96"))
```

Comment:
I see, you inherited this convoluted strlen(). Did I write that 20 years ago?  Maybe.  Thanks for improving upon this.